core/src/main/java/org/apache/druid/timeline/VersionedIntervalTimeline.java

```diff
@@ -849,4 +851,37 @@ public int hashCode()
       return Objects.hash(trueInterval, version, partitionHolder);
     }
   }
+
+  public static class PartitionChunkEntry<VersionType, ObjectType>
+  {
+    private final Interval interval;
```

Comment:
Please add javadoc explaining what this interval is. There are two types of interval in timeline. See `LogicalSegment`.